torch/csrc/autograd/python_function.cpp

```diff
@@ -471,71 +436,26 @@ static void _save_variables(THPFunction* self, t2var_type &t2var)
   self->saved_variables.reserve(num_saved);
   auto cdata_ptr = &self->cdata;
   for (int i = 0; i < num_saved; i++) {
-    PyObject *tensor = PyTuple_GET_ITEM(self->to_save, i);
-    if (tensor == Py_None) {
+    PyObject *obj = PyTuple_GET_ITEM(self->to_save, i);
+    if (obj == Py_None) {
       self->saved_variables.emplace_back();
       continue;
+    } else if (THPVariable_Check(obj)) {
+      auto variable = (THPVariable*)obj;
+      bool is_output = variable->cdata.grad_fn().get() == cdata_ptr;
+      self->saved_variables.emplace_back(variable->cdata, is_output);
+    } else if (THPModule_isTensor(obj)) {
+      // TODO: remove once Variable and Tensor classes are merged
```

Comment:
Can we make something like `// TODO [VarTensor]` so that we have a single thing to grep for after we merge them? There are a few places in this file that could be simplified, and they have a similar comment, but it will be hard to find them later

Reply:
We already have a lot of comments with this pattern. I don't think adding this convention will make it any easier to find those.